### PR TITLE
InfluxDb output : allow reusing writer

### PR DESCRIPTION
**Pull request by hightea**
Add equals and hashcode on InfluxDbWriterFactory to allow sharing the writer between servers and queries

### CODE REVIEW

**Review by gquintana — CHANGES_REQUESTED** [review]
[ERROR] Failed to execute goal org.pitest:pitest-maven:1.2.2:mutationCoverage (pitest) on project jmxtrans-output-influxdb: Mutation score of 26 is below threshold of 45 -> [Help 1]

File: jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriterFactory.java [hunk 1]
```diff
@@ -44,6 +45,7 @@
 import static java.lang.Boolean.FALSE;
 import static java.lang.Boolean.TRUE;
 
+@EqualsAndHashCode(exclude = {"influxDB"})
```
[the remark this answers is not in the document]
Comment (hightea) [reply jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriterFactory.java:48]: In the ServerListBuilder, the reuse of existing outputWriterFactory is made with the SingletonOutputWriterFactory.
The unicity ("singleton") of the SingletonOutputWriterFactory is based on the equals / hashcode of thisi class which is delegated to the equals / hashcode  of the outputWriterFactory. So defining these methods on the OutputWriterFactory allow the reuse of the factory between several queries / servers.

**Comment by hightea** [comment]
I'm working on the org.pitest:pitest-maven:1.2.2:mutationCoverage (pitest)  failure : 
This simply need me to add some tests to the InfluxDbWriterFactory... work in progress

File: jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriterFactory.java [hunk 2]
```diff
@@ -54,39 +59,51 @@
 	 * The retention policy</a> for each measuremen where no retentionPolicy
 	 * setting is provided in the json config
 	 */
-	private static final String DEFAULT_RETENTION_POLICY = "autogen";
+	@VisibleForTesting
+	static final String DEFAULT_RETENTION_POLICY = "autogen";
 
+	@Getter(AccessLevel.PACKAGE)
```
[the remark this answers is not in the document]
Comment (hightea) [reply jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriterFactory.java:65]: Only to have some access to data to do some tests...

**Pull request merged**